src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.PreserveCompilationContext.targets

```diff
@@ -73,4 +73,32 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        _CopyReferenceOnlyAssembliesForBuild
+
+    Copies reference assemblies that normally can't be resolved at runtime to the 'refs' folder in the build output.
+    This is necessary in order for the running app to resolve these reference assemblies.
+    ============================================================
+    -->
+  <Target Name="_CopyReferenceOnlyAssembliesForBuild"
```

Comment:
I modeled this target after [`CopyFilesToOutputDirectory`](https://github.com/Microsoft/msbuild/blob/a61828925261696fde4c57d0a809de06e1d9bad3/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4021), which doesn't set Inputs/Outputs.  Instead, each of the Copy tasks doesn't actually do the copy, since the file is unchanged and `$(SkipCopyUnchangedFiles)` is `true`.